Cython/Compiler/Symtab.py

```diff
@@ -697,6 +697,25 @@ def declare_enum(self, name, pos, cname, typedef_flag,
         self.sue_entries.append(entry)
         return entry
 
+    def declare_scoped_enum(self, name, pos, cname, create_wrapper = 0):
+        if name:
+            if not cname:
+                cname = name
+        if self.is_cpp_class_scope:
+            namespace = self.outer_scope.lookup(self.name).type
+        else:
+            namespace = None
+
+        entry = self.lookup_here(name)
+        if not entry:
```

Comment:
This suggests an error test (in `tests/errors/` with `# mode: error`) that a redeclaration gets detected and rejected.

Reply:
Thanks for catching this - I added an error test